src/librustc_resolve/lib.rs

```diff
@@ -2273,7 +2273,13 @@ impl<'a> Resolver<'a> {
 
                 if items.len() == 0 {
                     // Resolve prefix of an import with empty braces (issue #28388).
-                    self.smart_resolve_path(id, None, &path, PathSource::ImportPrefix);
+                    self.smart_resolve_path_with_crate_lint(
+                        id,
+                        None,
+                        &path,
+                        PathSource::ImportPrefix,
+                        CrateLint::SimplePath(id), // TODO seems wrong
```

Comment:
whoops forgot about TODO. I'll fix tomorrow. I couldn't actually reproduce a bug here, so maybe it's fine as is, but it *seems* like this is wrong in some case or other and we should be tracking the root-id and root-path

Reply:
Hm so this may fall in this category?

```rust
#![crate_type = "rlib"]
#![feature(rust_2018_preview)]
#![warn(rust_2018_compatibility)]
#![allow(unused_imports, dead_code)]

mod foo {}
mod bar {}


fn main() {
    use {foo::{}, bar::{}};
}
```

Even with https://github.com/rust-lang/rust/pull/50982 I think I'm still getting two warnings here

Reply:
I fixed it in any case.

Reply:
I found some other ways for it to go wrong, with a bit more effort =)